lib/engines/content_object_store/test/integration/content_block_editions_test.rb

```diff
@@ -57,6 +57,53 @@ class ContentBlockEditionsTest < ActionDispatch::IntegrationTest
     assert_equal details, new_edition.details
     assert_equal new_edition.content_block_document_id, new_document.id
   end
+
+  test "#create posts the new edition to the Publishing API" do
+    stub_request_for_schema("block_type")
+
+    document_title = "Some Title"
+    block_type = "block_type"
+    details = {
+      foo: "Foo text",
+      bar: "Bar text",
+    }
+    fake_put_content_response = GdsApi::Response.new(
+      stub("http_response", code: 200, body: {}),
+    )
+
+    # This UUID is created by the database so instead of loading the record
+    # we stub the initial creation so we know what UUID to check for.
+    ContentObjectStore::ContentBlockEdition.any_instance.stubs(:create_random_id)
+      .returns("49453854-d8fd-41da-ad4c-f99dbac601c3")
+
+    publishing_api_mock = Minitest::Mock.new
+    publishing_api_mock.expect :put_content, fake_put_content_response, [
+      "49453854-d8fd-41da-ad4c-f99dbac601c3",
+      {
+        schema_name: "content_block_type",
+        document_type: "content_block_type",
+        publishing_app: "whitehall",
+        title: "Some Title",
+        details: {
+          "foo" => "Foo text",
+          "bar" => "Bar text",
+        },
+      },
+    ]
+
+    Services.stub :publishing_api, publishing_api_mock do
+      ContentObjectStore::ContentBlockSchema.stub_const(:VALID_SCHEMAS, [block_type]) do
```

Comment:
Do we need to do `stub_const` here now, or can we just stub `ContentObjectStore::ContentBlockSchema.valid_schemas` now?

Reply:
Ignore me, I can see this was fixed in the next commit.